### PR TITLE
#10214: Mobile issue - Overlay of tools and info

### PR DESCRIPTION
**Pull request by mahmoudadel54**
## Description
In this PR, a custom class for leaflet scale bar is adding if the widgets tray plugin is existing on map for mobile mode

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [x] Bugfix
 - [ ] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [ ] Other... Please describe:

## Issue
#10214 

**What is the current behavior?**
#10214 

**What is the new behavior?**
The UI in case of existing widgets tray button on mobile mode: 
-  image
-  image

The UI in case of normal mobile mode without the widgets tray button:
-  image

- image

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

### CODE REVIEW

File: web/client/components/map/leaflet/ScaleBar.jsx [hunk 1]
```diff
@@ -41,6 +43,14 @@ class ScaleBar extends React.Component {
...
                 }
             }
+            // as there is no direct way to set a custom class to scalebar in leaflet -->
+            // use DOM munipulating to get scalebar element and set the needed custom class that holds your style
+            // here the needed for mobile mode in case of widgetsTray widget is existing on map
+            const widgetsTrayBtnExistInMobileMode = this.props.isMobile && document.querySelector('.widgets-tray');
+            if (widgetsTrayBtnExistInMobileMode) {
+                const scaleBarElement  = document.querySelector('.leaflet-control-scale');
+                scaleBarElement .classList.add('mobile-map-with-widgets-tray');
+            }
```
Comment (allyoucanmap) [inline web/client/components/map/leaflet/ScaleBar.jsx:53]: We should avoid dom manipulation like this one, in particular we cannot bind an hardcoded class of a different component inside another one.

My proposal is to remove this part and related class and move the logic to the [WidgetTray](https://github.com/geosolutions-it/MapStore2/blob/master/web/client/plugins/widgets/WidgetsTray.jsx) plugin bacause it is already connected to the state and we could move the margin bottom based on some condition:

- if it is mobile device and not 3D visualization apply a margin bottom of 60
- other cases use the default value, currently 32

Expected results (cc @tdipisa )

mobile 2D
image

mobile 2D with widgets
image

mobile 3D with widgets
image

File: web/client/plugins/Map.jsx [hunk 2]
```diff
@@ -371,7 +371,8 @@ class MapPlugin extends React.Component {
         return this.props.tools.map((tool) => {
             const Tool = this.getTool(tool);
             const options = this.props.toolsOptions[Tool.name] && this.props.toolsOptions[Tool.name][this.props.mapType] || this.props.toolsOptions[Tool.name] || {};
-            return <Tool.impl key={Tool.name} {...options}/>;
+            const isMobile = ConfigUtils.getBrowserProperties().mobile;
+            return <Tool.impl key={Tool.name} {...options} isMobile={isMobile} />;
```
Comment (allyoucanmap) [inline web/client/plugins/Map.jsx:375]: please remove this based on previous comment

File: web/client/themes/default/less/leaflet.less [hunk 3]
```diff
@@ -71,11 +71,14 @@
...
-    top: 10px;
-    right: 10px;
+    bottom: 5px;
+    right: 45px ;
     clear: none;
     z-index: 1000;
 }
+.leaflet-control-scale.mobile-map-with-widgets-tray {
+    right: 95px;
+}
```
Comment (allyoucanmap) [inline web/client/themes/default/less/leaflet.less:81]: please remove this based on previous comment

File: web/client/themes/default/less/leaflet.less [hunk 4]
```diff
@@ -71,11 +71,14 @@
 }
 
 .leaflet-control-scale {
-    top: 10px;
-    right: 10px;
+    bottom: 5px;
+    right: 45px ;
```
Comment (allyoucanmap) [inline web/client/themes/default/less/leaflet.less:75]: Now we could apply a right value of 55px so widget tray and scale bar are aligned

**Review by allyoucanmap — CHANGES_REQUESTED** [review]

File: web/client/components/map/leaflet/ScaleBar.jsx (hunk 1 above)
Comment (tdipisa) [reply web/client/components/map/leaflet/ScaleBar.jsx:53]: ok that's also fine @allyoucanmap

**Review by allyoucanmap — APPROVED** [review]

**Comment by allyoucanmap** [comment]
@ElenaGallo please test on dev and let us know if we can backport, thanks

**Comment by ElenaGallo** [comment]
Test passed, @mahmoudadel54 please backport to 2024.01.xx. Thanks

**Comment by mahmoudadel54** [comment]
> Test passed, @mahmoudadel54 please backport to 2024.01.xx. Thanks

Backport is Done --> https://github.com/geosolutions-it/MapStore2/pull/10313

**Pull request merged**